pennylane/ops/op_math/prod.py

```diff
@@ -422,6 +445,50 @@ def _sort(cls, op_list, wire_map: dict = None) -> List[Operator]:
 
         return op_list
 
+    def terms(self):
+        r"""Representation of the operator as a linear combination of other operators.
+
+        .. math:: O = \sum_i c_i O_i
+
+        A ``TermsUndefinedError`` is raised if no representation by terms is defined.
+
+        Returns:
+            tuple[list[tensor_like or float], list[.Operation]]: list of coefficients :math:`c_i`
+            and list of operations :math:`O_i`
+
+        ** Example **
+
+        >>> qml.operation.enable_new_opmath()
+        >>> op = X(0) @ (0.5 * X(1) + X(2))
+        >>> op.terms()
+        ([0.5, 1.0],
+         [X(1) @ X(0),
+          X(2) @ X(0)])
+
+        """
+        # try using pauli_rep:
+        if pr := self.pauli_rep:
```

Comment:
you mean caching `Prod.terms()` or `Prod.pauli_rep`?

Reply:
Some ballpark numbers:
```
from pympler import asizeof

data = qml.data.load("qchem", molname="C2H2", bondlength=1.203, basis="STO-3G")[0]
H2 = data.hamiltonian
H = qml.dot(*H2.terms())

asizeof.asizeof(H.terms()), asizeof.asizeof(H.pauli_rep)
```
```
(9204088, 3828176)
```

These are in bytes, so 9.2MB and 3.8MB, respectively

Reply:
Good points. Thanks for the stats, that changed my mind!

Reply:
Ooff. We should probably think about the memory footprint more going forward.

Reply:
Yeah I am actually really surprised by these numbers 🤔

Reply:
These are quite compressible though, at least the terms() in form of numpy arrays in npy format are just 50KB whereas the pauli rep is 631KB (unsurprisingly more since its a dict and not an array)